### ISSUE TITLE
Calling `redirect()` on a page (server-side) causes hydration error in a browser.

### ISSUE THREAD

**Issue by dream-ellie** [vercel/next.js#47146]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: Ubuntu 20.04.0 LTS Wed Mar 15 2023 17:46:37 GMT+0900 (Korean Standard Time)
    Binaries:
      Node: 16.14.2
      npm: 7.17.0
      Yarn: 1.22.19
      pnpm: 7.13.6
    Relevant packages:
      next: 13.2.5-canary.3
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

https://stackblitz.com/edit/vercel-next-js-cygjqi?file=app/page.tsx

### To Reproduce

1. Open the link to the StackBlitz given above
2. run `yarn build` then `yarn start`
3. Navigate a few times between "Home" and "About" links that are rendered by the example app.
4. Open browser developer tools window and console drawer.
5. Observe the errors.

### Describe the Bug

**TL;DR**
🔥 Hydration error is logged in a browser console if a page (server-component) contains call to `redirect('url')` method

![Screen Shot 2023-03-15 at 5 50 40 PM](https://user-images.githubusercontent.com/61736137/225256348-1ec401ae-070b-40c8-935a-c94b9d250620.png)

**Details**

The code like below will cause hydration error in a browser:
👇
**/page.tsx**

```
import { redirect } from 'next/navigation';

export default function Home() {
  redirect('/auth');

  return <p>HOME PAGE</p>;
}
```
👉 **_Note that this only happens in production mode and does not produce any errors in development, which makes it harder to debug because minified error has no useful information._**

### Expected Behavior

There should be no error in the browser console.

### Which browser are you using? (if relevant)

chrome

### How are you deploying your application? (if relevant)

vercel

**Comment by Fredkiss3** [comment]
It is not a hydration error, it is a error that the redirect function throws in order for next to do the redirect. You don't have to handle this error by yourself or mind about it, since it does not change the behavior of your app in any way. 

A hydration usually contains an error code and a link to the reactjs website : 
<img width="259" alt="image" src="https://user-images.githubusercontent.com/38298743/225278815-b8bd3d57-8bda-484c-847e-93128596ab64.png">

**Comment by Fredkiss3** [comment]
Sorry what i meant to say is that the redirection still happens and it will work normally, it is just the error log that should not be there. There is a similar issue still open for more context : #42556 and a comment on why it does that : https://github.com/vercel/next.js/issues/42556#issuecomment-1305366461

**Comment by dream-ellie** [comment]
> @JoelBonet @dream-ellie what i meant to say is that the redirection still happens and it will work normally, it is just the error log that should not be there. There is a similar issue still open for more context : #42556 and a comment on why it does that : [#42556 (comment)](https://github.com/vercel/next.js/issues/42556#issuecomment-1305366461)

Thanks @Fredkiss3! I suspected this is some sort of mistreatment of the `NEXT_REDIRECT` that is thrown on redirect (according to the docs) and was not sure if it impacts anything else besides the error log in the console. 
Indeed, functionality-wise I have not noticed anything broken.

From the https://github.com/vercel/next.js/issues/42556#issuecomment-1305366461 I see that the  error message will be taken care of eventually, so I think this ticket can be duplicated to https://github.com/vercel/next.js/issues/42556 and closed.

**Comment by balazsorban44** [comment]
Indeed, this has been fixed on `next@canary` already, so closing the issue! Please upgrade!

**Comment by y0zong** [comment]
just tested `13.3.1-canary.8` still not fix this, in prod this error case route to 404 and this is a serious mistake
